app/views/courses/StudentRankingView.coffee

```diff
@@ -0,0 +1,198 @@
+helper = require 'lib/coursesHelper'
+require('app/styles/courses/student-ranking-view.sass')
+RootView = require 'views/core/RootView'
+template = require 'templates/courses/student-ranking-view'
+LevelSession = require 'models/LevelSession'
+CocoCollection = require 'collections/CocoCollection'
+Campaign = require 'models/Campaign'
+Level = require 'models/Level'
+utils = require 'core/utils'
+require 'three'
+UserPollsRecord = require 'models/UserPollsRecord'
+Poll = require 'models/Poll'
+CourseInstance = require 'models/CourseInstance'
+api = require 'core/api'
+Classroom = require 'models/Classroom'
+Course = require 'models/Course'
+CourseInstance = require 'models/CourseInstance'
+Levels = require 'collections/Levels'
+store = require 'core/store'
+
+require('vendor/scripts/jquery-ui-1.11.1.custom')
+require('vendor/styles/jquery-ui-1.11.1.custom.css')
+fetchJson = require 'core/api/fetch-json'
+Users = require 'collections/Users'
+
+require 'lib/game-libraries'
+
+class LevelSessionsCollection extends CocoCollection
+  url: ''
+  model: LevelSession
+
+  constructor: (model, student) ->
+    super()
+    @url = "/db/user/#{student}/level.sessions?project=state.complete,levelID,state.difficulty,playtime,state.topScores,codeLanguage,level"
+
+class CampaignsCollection extends CocoCollection
+  # We don't send all of levels, just the parts needed in countLevels
+  url: '/db/campaign/-/overworld?project=slug,adjacentCampaigns,name,fullName,description,i18n,color,levels'
+  model: Campaign
+
+module.exports = class StrudentRankingView extends RootView
+  id: 'campaign-view'
+  template: template
+
+  
+  constructor: (options, @terrain) ->
+
```

Comment:
Should we do another check on this page to ensure that only users of the classroom can access this page?
And redirect other users back to home.

Reply:
Add on the next commit.